rados/read_op_omap_get_vals_by_keys.go

```diff
@@ -0,0 +1,126 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package rados
+
+// #cgo LDFLAGS: -lrados
+// #include <rados/librados.h>
+// #include <stdlib.h>
+//
+import "C"
+
+import (
+	"unsafe"
+
+	"github.com/ceph/go-ceph/internal/cutil"
+)
+
+// ReadOpOmapGetValsByKeysStep holds the result of the
+// GetOmapValuesByKeys read operation.
+// Result is valid only after Operate() was called.
+type ReadOpOmapGetValsByKeysStep struct {
+	withRefs
+
+	// C arguments
+
+	cKeys cutil.CPtrCSlice
+	iter  C.rados_omap_iter_t
+	prval *C.int
+
+	// Internal state
+
+	// canIterate is only set after the operation is performed and is
+	// intended to prevent premature fetching of data
+	canIterate bool
+}
+
+func newReadOpOmapGetValsByKeysStep(keys []string) *ReadOpOmapGetValsByKeysStep {
+	s := &ReadOpOmapGetValsByKeysStep{
+		cKeys: cutil.NewCPtrCSlice(len(keys)),
+		prval: (*C.int)(C.malloc(C.sizeof_int)),
+	}
+
+	for i, key := range keys {
+		cKey := C.CString(key)
+		s.add(unsafe.Pointer(cKey))
+		s.cKeys[i] = cutil.CPtr(cKey)
+	}
+
+	return s
+}
+
+func (s *ReadOpOmapGetValsByKeysStep) free() {
+	s.canIterate = false
+	C.rados_omap_get_end(s.iter)
+
+	s.cKeys.Free()
+	s.cKeys = nil
+
+	C.free(unsafe.Pointer(s.prval))
+	s.prval = nil
+
+	s.withRefs.free()
+}
+
+func (s *ReadOpOmapGetValsByKeysStep) update() error {
+	err := getError(*s.prval)
+	s.canIterate = (err == nil)
+
+	return err
+}
+
+// Next gets the next omap key/value pair referenced by
+// ReadOpOmapGetValsByKeysStep's internal iterator.
+// If there are no more elements to retrieve, (nil, nil) is returned.
+// May be called only after Operate() finished.
+//  PREVIEW
+func (s *ReadOpOmapGetValsByKeysStep) Next() (*OmapKeyValue, error) {
+	if !s.canIterate {
+		return nil, ErrOperationIncomplete
+	}
+
+	var (
+		cKey    *C.char
+		cVal    *C.char
+		cValLen C.size_t
+	)
+
+	ret := C.rados_omap_get_next(s.iter, &cKey, &cVal, &cValLen)
```

Comment:
As you like. You are welcome to change to the *2 versions in a follow up commit. 😉

Reply:
I'd rather do it separately

Reply:
Doing it separately is A-OK with me.  I highly encourage you to do it soon because I don't love the idea of adding new code based on deprecated functions (assuming I am reading this discussion correctly). I am not asking that you do it before the release, just do it before we all forget. :-)